collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,346 @@
+
+We should generate an empty package.json file which can be edited by developer as and when needed.
+
+###### index.js
+
+We should provide `index.js`(which would start the server and serve the handler function) out of the box which can be used by the developer to test his code locally.
+
+###### readme
+
+We should also provide a readme.md which would consist of following:
+
+* Instructions how he can deploy the function into the cluster.
+* Instructions how he can run the function locally.
+* Instructions on how to use the `config.yaml` file
+* Instructions on various folder structure.
+* Instructions on how to test debug it locally like using telepresence.
+
+##### Testing/debugging it locally
+
+We can introduce commands to test the function locally and also if the developer wants to debug the code. This is useful in case when we have some event payloads that are not easy to recreate.
+
+`kyma function debug <function-name> -n <namespace>`
+
+> We can use telepresence for local debugging. We can either use --swap-deployment or --new-deployment option.
+
+Telepresence would create a proxy in the kyma cluster and redirect the calls to local codebase. One can attach the debugger to his code base along with breakpoints to debug the code.
+
+##### Deploying and running the project
+
+The function generated by `init` command should be deployable onto kyma cluster. For deploying the file onto kyma cluster it would read the values from `config.yaml`. If the api name is not set then we should show some warning message and deploy.
+
+For running/debugging it locally we should provide makefile. The running/debugging commands should be part of readme.
+
+#### Working with deployed kyma cluster
+
+We should allow deployment of the functions on a k8s cluster. Following commands should be supported
+
+##### CRUD
+
+##### Commands that use config
+
+`kyma function create -p <path-to-function-dir>`
+> This would create a function after reading the `config.yaml` from the path provided and to read the config like events, api etc.
+
+`kyma function update -p <path-to-function-dir>`
+> Update an existing function with the `config.yaml` present in the directory.
+
+##### Expanded command with switches
+
+`kyma function create <function-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma function update <function-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma function delete <function-name> -n <namespace>`
+
+`kyma function get <function-name> -n <namespace>`
+
+*Labels*
+
+`kyma function label <function-name> ["foo=bar",..] -n <namespace>`
+
+*Expose*
+
+`kyma function trigger <function-name> --http --secure(optional) -n <namespace>`
```

Comment:
I think we said `expose` would be better here.

Reply:
fixed.